### PR TITLE
fix: all data in one call

### PR DESCRIPTION
**Pull request by Okavango**

### CODE REVIEW

File: contracts/pool/constant-product/ConstantProductPoolFactory.sol [hunk 1]
```diff
@@ -29,4 +37,39 @@ contract ConstantProductPoolFactory is PoolDeployer {
         pool = address(new ConstantProductPool{salt: salt}(_deployData, masterDeployer));
         _registerPool(pool, tokens, salt);
     }
+
+    // @dev tokens MUST be sorted i < j => token[i] < token[j]
+    // @dev tokens.length < 256
+    function getPoolsForTokens(address[] calldata tokens)
```
Comment (sarangparikh22) [inline contracts/pool/constant-product/ConstantProductPoolFactory.sol:43]: I would prefer keeping the Factory minimal and use a separate Helper contract instead for this calculation.

Comment (Okavango) [reply contracts/pool/constant-product/ConstantProductPoolFactory.sol:43]: I see your point. What others think about it?

Comment (Okavango) [reply contracts/pool/constant-product/ConstantProductPoolFactory.sol]: OK, have moved to the separate contract

Comment (Okavango) [reply contracts/pool/constant-product/ConstantProductPoolFactory.sol]: @sarangparikh22 What do you think?

Comment (sarangparikh22) [reply contracts/pool/constant-product/ConstantProductPoolFactory.sol]: Yeah looks good to me @Okavango

Comment (sarangparikh22) [reply contracts/pool/constant-product/ConstantProductPoolFactory.sol]: Yeah looks good to me @Okavango

**Review by sarangparikh22 — APPROVED** [review]

**Pull request merged**